version: '3.9'

services:
    etl:
      container_name: etl
      restart: always
      build: ./tgg
      ports:
        - "7501:7501"
      environment:
        - DATABASE_URI=postgresql://gestion_guarico:MepmD1VSbEIDB6ncJNzLjwFsk5UgVa@db:5432/DATABASE
        - PRODUCTION=False
        - TZ=America/Caracas
      depends_on:
        - db
      # command: gunicorn -w 2 -b :7501 --timeout 1200 app:app
      command: bash -c "alembic upgrade head && python main.py" 
      networks:
        - postgres-network


    db:
      image: postgres:14.5
      container_name: postgresql_Metabase_db
      hostname: db
      restart: always
      environment:
        - POSTGRES_PASSWORD=MepmD1VSbEIDB6ncJNzLjwFsk5UgVa
        - POSTGRES_USER=gestion_guarico
        - POSTGRES_DB=DATABASE
      ports:
        - 7503:5432
      volumes:
        - ./db-data:/var/lib/postgresql/data
        - ./db:/docker-entrypoint-initdb.d/
      networks:
        - postgres-network

    metabase:
      image: metabase/metabase:v0.44.2
      container_name: postgresql_Metabase_web
      volumes:
        - ./metabase-data:/metabase-data
      environment:
        - MB_DB_FILE=/metabase-data/metabase.db
        - MB_DB_TYPE=postgres
        - MB_DB_DBNAME=DATABASE
        - MB_DB_PORT=5432
        - MB_DB_USER=gestion_guarico
        - MB_DB_PASS=MepmD1VSbEIDB6ncJNzLjwFsk5UgVa
        - MB_DB_HOST=db
      ports:
        - 7502:3000
      restart: always
      depends_on: 
        - db
      networks:
        - postgres-network
networks:
  postgres-network:
    driver: bridge
